var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -41,12 +41,14 @@ class Openmpi(Package):
     def url_for_version(self, version):
         return "http://www.open-mpi.org/software/ompi/v%s/downloads/openmpi-%s.tar.bz2" % (version.up_to(2), version)
 
-    def setup_dependent_environment(self, module, spec, dep_spec):
-        """For dependencies, make mpicc's use spack wrapper."""
-        os.environ['OMPI_CC'] = 'cc'
-        os.environ['OMPI_CXX'] = 'c++'
-        os.environ['OMPI_FC'] = 'f90'
-        os.environ['OMPI_F77'] = 'f77'
+    def environment_modifications(self, dependent_spec):
+        env = super(Openmpi, self).environment_modifications(dependent_spec)
+        # FIXME : the compilers should point to the current wrappers, not to generic cc etc.
```

Comment:
This gets complicated.  These variables are intended to be set to *Spack's* wrappers during a Spack build, so that dependents using MPI implementations properly RPATH their dependencies.  I have recently put in changes that let you refer to these more easily, so you could do this:

```python
        env.set_env('OMPI_CC', spack_cc)
        env.set_env('OMPI_CXX', spack_cxx)
        env.set_env('OMPI_FC', spack_fc)
        env.set_env('OMPI_F77', spack_f77)
```

However, if they are loaded by a module *outside of spack*, they should probably point to the real compilers.  So you want to point at these:

```python
        env.set_env('OMPI_CC', self.compiler.cc)
        env.set_env('OMPI_CXX', self.compiler.cxx)
        env.set_env('OMPI_FC', self.compiler.fc)
        env.set_env('OMPI_F77', self.compiler.f77)
```

This could be made more consistent/intuitive (sorry!) but that's what you want at the moment.  Maybe we could add `real_cc`, `real_cxx`, or similar later, but let's ignore that for now.

I'm not sure how best to do this -- the EnvModifications *may* need a way to indicate that different things should be done inside Spack and outside.  Or maybe there is a cleaner way to do it... I *think* this is only an issue for things that wrap compilers, or at least that is where I have encountered it so far.  Suggestions welcome!